pkg/cloudproviders/azure/metadata.go

```diff
@@ -79,5 +85,28 @@ func GetMetadata(ctx context.Context) (*storage.ProviderMetadata, error) {
 			},
 		},
 		Verified: verified,
+		Cluster:  clusterMetadata,
 	}, nil
 }
+
+func getClusterMetadata(ctx context.Context) *storage.ClusterMetadata {
+	k8sClient, err := k8sutil.GetK8sInClusterClient()
+	if err != nil {
+		log.Error("Failed to kubernetes client: ", err)
+		return &storage.ClusterMetadata{}
```

Comment:
(applies to other places as well): why not return nil in case of the default value for `ClusterMetadata`?